sdk/include/opentelemetry/sdk/trace/always_off_sampler.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/sampler.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+/**
+ * The always off sampler always returns NOT_RECORD, effectively disabling
+ * tracing functionality.
+ */
+class AlwaysOffSampler : public Sampler
+{
+public:
+  /**
+   * @return Returns NOT_RECORD always
+   */
+  SamplingResult ShouldSample(
+    const SpanContext *parent_context,
+    trace_api::TraceId trace_id,
+    nostd::string_view name,
+    trace_api::SpanKind span_kind,
+    const trace_api::KeyValueIterable &attributes) noexcept override
```

Comment:
We try to be consistent in [not naming unused parameters](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f9-unused-parameters-should-be-unnamed). See [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/trace/noop.h#L38) for an example.